### DISCUSSION TITLE
wie kann ich es verdrahten und installieren

### DISCUSSION THREAD

**Discussion by ap20132022** [Trunks1982/Daikin-Rotex-HPSU-CAN#6] [Q&A] [answered]
Hallo zusammen,
ich hab' noch keine Erfahrung mit esphome gehabt. 
Folgendes hab' ich gemacht:
- auf https://web.esphome.io/ gegangen, mit dem Gerät verbinden
- auf "Prepare for first use" klicken
- nach der Installation --> klicken auf "install" und die Datei [Rotex-Daikin-CAN.bin] ausgewählt --> nochmal installieren
- aktuell bekomme ich unter "logs" nur: invalid header
Ich verwende Esp32 Cam
Wie kann ich die GPIO47 & 48 in andere GPIO ändern? 
Ich stelle mir vor, dass der nächster Schritt ist, die yaml Datei in Home assistant zu importieren.

Bitte um eure Unterstüzung und bedanke mich herzlich dafür.

**Comment by Trunks1982** [comment]
Hi hast du webui zugriff auf den ESP?
Hast du den Transreciever richtig angeschlossen?
Kannst du ein Screenshot davon schicken?

**Reply by ap20132022** [reply]
vielen DANK für deine rasche Antwort. 
Ich befürchte, dass es mit meinem EspCam nicht funktioniert. 
Nach "Prepare for first use" bekomme ich nur: An error occurred. Improv Wi-Fi Serial not detected
Einen ausführlichen Log hab' ich bei gefügt. 
[logs.txt](https://github.com/Trunks1982/Daikin-Rotex-HPSU-CAN/files/13985043/logs.txt)
Ich komme gar nicht zu dem webui.

**Reply by Trunks1982** [reply]
Meine bin ist für einen S3, wenn es keiner ist sollte das dann auch nicht funktionieren.Er versucht aufjedenfall im log das Wlan zu starten.Normalerweise sollte dich dein Hotspot nach 5min aufbauen.
Ist wahrscheinlich auch nicht der Fall oder?

**Comment by ap20132022** [comment]
es zeigt kein Hotspot an. Daher kann ich es nicht (innerhalb 5 min.) verbinden.
Ich hab' mit einem WR32ROM (mit der "Prepare for first use") ausprobiert. Dieses lässt sich sofort mit meinem Wlan verbinden.
Ich bestelle mir mal ein S3 und probiere nochmal. 
Dir einen schönen Tag, schönes WE irgendwann. DANKE für die Hilfe.

**Reply by Trunks1982** [reply]
Mach das.Lass mich mal wissen wie es ausgegangen ist.

**Comment by ap20132022** [comment]
Hallo Trunks1982,
ich brauche deine Hilfe. 
Heute ist meine ESP S3 N16R8 angekommen. 
Mit ESPHome konnte ich "install for first use" durchführen. Anschließend hab' ich deine .bin Datei installiert. 
Dann ist mein ESP nicht mehr erreichbar. Aktuell hab' ich das CAN-Modul sowie meine WP noch nicht angeschlossen. 
So sieht ein Teil der Log File aus:
`ESP-ROM:esp32s3-20210327
Build:Mar 27 2021
rst:0x7 (TG0WDT_SYS_RST),boot:0x8 (SPI_FAST_FLASH_BOOT)
Saved PC:0x400454d5
SPIWP:0xee
mode:QIO, clock div:1
load:0x3c0c0020,len:0x23d08
ets_loader.c 78 
ESP-ROM:esp32s3-20210327
Build:Mar 27 2021
rst:0x7 (TG0WDT_SYS_RST),boot:0x8 (SPI_FAST_FLASH_BOOT)
Saved PC:0x400454d5
SPIWP:0xee
mode:QIO, clock div:1
load:0x3c0c0020,len:0x23d08
ets_loader.c 78 
ESP-ROM:esp32s3-20210327
Build:Mar 27 2021
rst:0x7 (TG0WDT_SYS_RST),boot:0x8 (SPI_FAST_FLASH_BOOT)
Saved PC:0x400454d5
SPIWP:0xee
mode:QIO, clock div:1
load:0x3c0c0020,len:0x23d08
ets_loader.c 78 `
[logs (1).txt](https://github.com/Trunks1982/Daikin-Rotex-HPSU-CAN/files/14097837/logs.1.txt)

Brauche ich unbedingt das CAN-Modul anzuschiließen? Was hab' ich falsch gemacht?
VIELEN DANKE für deine HIlfe.

**Reply by Trunks1982** [reply]
Meine Meinung nach muss das  canmodul angeschlossen werden.Er will ja sofort die gpios ansprechen.Hast du denn einen Hotspot vom esp gehabt?
Du kannst meine bin auch direkt als erstes flashen.

**Reply by ap20132022** [reply]
ich hab' dein bin Datei direkt geflasht. Hab' auch mit dem WP über CAN angeschlossen. Es hat keine Veränderung.
Liegt es vielleicht an das mein Modul?
Folgendes Modul hab' ich verwendet:
![image](https://github.com/Trunks1982/Daikin-Rotex-HPSU-CAN/assets/122316058/c10d3992-46a0-4eb4-bc29-bc7501074d7c)

**Reply by Trunks1982** [reply]
Müsste eigentlich gehn ist jetzt nur die frage wegen den pins.

**Reply by Trunks1982** [reply]
Hast ne möglichkeit esphome zuverwenden?

**Reply by ap20132022** [reply]
Hab Pin 47 & 48 verwendet. 3,3V Spannung + GND.
Was meinst du mit ESP Home? Hab über web und auch über Home Assistant ausprobiert. Nach Upload deine bin Datei ist es nicht mehr über WLAN erreichbar

**Reply by Trunks1982** [reply, marked as answer]
Wenn du homeassist hat installier mal esphome.Kannst dann auch die yaml dann einfügen und dierekt dein wlan eingeben.

**Reply by ap20132022** [reply]
Jetzt hab' ich es geschafft. Herzlichen DANK für deine Hilfe. 
Hab' bisher Null-Ahnung von ESPhome. Es ist wirklich so einfach. Bin nur zu dumm dafür. 

![image](https://github.com/Trunks1982/Daikin-Rotex-HPSU-CAN/assets/122316058/e8b09fcb-963a-43c6-8e34-783fa2027d2f)

**Reply by Trunks1982** [reply]
Gut das du es sagst das es doch einfach ist.🤣 Bekommste jetzt werte?

**Reply by ldnrktml** [reply]
Woran lag es denn? Vllt. auch für andere interessant, die ein ähnliches Problem haben?
Hast du nun die *.bin "Standalone" Lösung oder yaml die über HomeAssistant genutzt?

**Reply by ap20132022** [reply]
ja (siehe Screenshot).
Einfach anzuwenden meine ich.
Wo hast du die CAN-Register her? Wollte schon ausprobieren, es zu erweitern.

**Reply by Trunks1982** [reply]
Aus iobroker.Bekommt man aber auch von der FHEM  Lösung von Github.

**Reply by ap20132022** [reply]
Ich hab zuerst die bin Datei verwendet. Es hat nicht funktioniert (aufgrund Modulunterschied???)
Anschließend hab ich ESP Home auf HA installiert. Es hat beim ersten Mal nicht über WLAN funktioniert (?). Keine Ahnung warum (schwaches Signal?)

Hab dann eine .bin Datei generiert und per USB auf dem Gerät geladen. 
Jetzt funktioniert es einwandfrei. Muss morgen Kabel/Gerät noch in Ordnung bringen. 
Überlege ob ich die Smartgrid Funktion rein programmieren. Eigentlich kann ich es auch über WW Temperatur anpassen.

**Reply by Trunks1982** [reply]
Über WW ist am einfachsten.

**Comment by ldnrktml** [comment]
@ap20132022: Nach dem flashen der bin hat der ESP einen neuen Hotspot, wie in der yaml angegeben. 
Dieser muss auf jeden Fall sichtbar sein und mit diesem kannst du dich verbinden und dann den ESP32 über die IP und dessen Webinterface in dein WLAN einbinden.

Nachteil der bin Lösung ist, dass man den GPIO nicht ändern kann. 

Über HomeAssistant kannst du es einfacher lösen, indem du die yaml über esphome (in HomeAssistant) flashst.
In dieser kann man auch die Daten des Hotspot, die Daten des eigenen WLAN und die GPIOs anpassen.

Ich selbst verwende einen ESP32 S3 Zero, anbei mal ein Foto mit CAN Interface an GPIO 2 und 3.

![espzerso](https://github.com/Trunks1982/Daikin-Rotex-HPSU-CAN/assets/99878337/95d4ae02-bc05-4c26-a7f8-e127504bf467)
